osquery/tables/applications/linux/docker_deb_packages.cpp

```diff
@@ -0,0 +1,192 @@
+      std::string value = line.substr(separator_position + 1, line.length());
+
+      auto it = kFieldMappings.find(key);
+      if (it != kFieldMappings.end()) {
+        boost::algorithm::trim(value);
+        w.Key(it->second.c_str());
+        w.String(std::move(value));
+      }
+    }
+    varbuf_reset(&vb);
+  }
+  varbuf_destroy(&vb);
+
+  w.EndObject();
+}
+
+void getDebPackages(int fd) {
+  if (!osquery::isDirectory(kDPKGPath)) {
+    TLOG << "Cannot find DPKG database: " << kDPKGPath;
+    return;
+  }
+
+  auto dropper = DropPrivileges::get();
+  dropper->dropTo("nobody");
+
+  rapidjson::StringBuffer buffer;
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+  writer.StartArray();
+
+  struct pkg_array packages;
+  dpkg_setup(&packages);
+
+  const auto* pk = packages.pkgs;
+  for (const auto& pkg : boost::make_iterator_range(pk, pk + packages.n_pkgs)) {
+    // Casted to int to allow the older enums that were embeded in the
+    // packages
+    // struct to be compared
+    if (static_cast<int>(pkg->status) !=
+        static_cast<int>(PKG_STAT_NOTINSTALLED)) {
+      extractDebPackageInfo(pkg, writer);
+    }
+  }
+
+  dpkg_teardown(&packages);
+  writer.EndArray();
+
+  const auto bufSize = buffer.GetSize();
+  if (bufSize >= 1) {
+    if (write(fd, buffer.GetString(), bufSize) != (ssize_t)bufSize) {
+      VLOG(1) << "error writing pkginfo ";
+    }
+  }
+}
+
+QueryData genDockerDebPackages(QueryContext& context) {
+  QueryData results;
+
+  for (const auto& id : context.constraints["id"].getAll(EQUALS)) {
+    if (!checkConstraintValue(id)) {
+      continue;
+    }
+
+    pt::ptree container;
+    Status s = dockerApi("/containers/" + id + "/json", container);
+    if (!s.ok()) {
+      VLOG(1) << "Error getting docker container info " << id << ": "
+              << s.what();
+      continue;
+    }
+    if (!container.get<bool>("State.Running", false)) {
+      continue;
+    }
+    pid_t pid = container.get<int>("State.Pid", 0);
+    if (pid == 0) {
+      VLOG(1) << "Error getting pid for container " << id;
+      continue;
+    }
+
+    int fds[2];
+    if (socketpair(AF_UNIX, SOCK_STREAM, 0, fds) == -1) {
+      VLOG(1) << "unable to open socket pair" << strerror(errno);
+      continue;
+    }
+
+    NamespaceOps nsOps(pid, fds[1]);
+    Status iv = nsOps.invoke(getDebPackages);
+    if (!iv.ok()) {
+      VLOG(1) << "error entering namespace: " << iv.what();
+      close(fds[0]);
+      close(fds[1]);
+      continue;
+    }
+
+    close(fds[1]);
+    FILE* fp = fdopen(fds[0], "rb");
+    if (fp == NULL) {
+      VLOG(1) << "unable to open fd " << strerror(errno);
+      close(fds[0]);
+      Status k = nsOps.kill();
+      if (!k.ok()) {
+        VLOG(1) << "unable to terminate foked process " << k.what();
+      }
+      continue;
+    }
+    char readBuffer[BUF_SIZE];
+    rapidjson::FileReadStream is(fp, readBuffer, sizeof(readBuffer));
+    rapidjson::Document d;
```

Comment:
nit: I know it's tempting to be terse with variables, but would you mind here, and throughout, being a _bit_ more descriptive? For instance `doc` instead of `d`?